### PR TITLE
Introduce BatcherHash version 1

### PR DESCRIPTION
**Pull request by roberto-bayardo**
This PR introduces a new BatcherHash version to allow the chain operator to configure whether the batcher can use Blobs to post batches.

We introduce this as a system-config contract option instead of, say, a batcher command-line config option, because the L2 GasPriceOracle contract needs to know what type of batcher transaction is being used so it can estimate the L1 data fee appropriately.

TODO: Unit tests

### CODE REVIEW

**Comment by roberto-bayardo** [comment]
Hmm, alternative implementation would be to just make the "CanBatcherUseBlobs" a member of an updated GasPriceOracle ...  the batcher can get at it through a regular L1 client call rather than chain-derivation pass through.

File: specs/system_config.md [hunk 1]
```diff
@@ -28,6 +28,11 @@ The first byte identifies the version.
 
 Version `0` embeds the current batch submitter ethereum address (`bytes20`) in the last 20 bytes of the versioned hash.
 
+Version `1`, which is allowed only following blob activation time, also embeds the batch submitter
+ethereum address (`bytes20`) in the last 20 bytes of the versioned hash. Additionally, the second
```
Comment (tynes) [inline specs/system_config.md:32]: I don't understand how this relates to the `GasPriceOracle`

**Comment by tynes** [comment]
The downside of this approach is that governance or the future chain gov entity would need to update the batcher hash value. Since fee markets are very dynamic, it would not be able to update in time for short changes. It would be more of a "only use this" sort of design. It also adds some complexity with the "sequencer allowlist" implementation in the future, since it would have to become aware of the version

**Comment by tynes** [comment]
I am much more in favor of this sort of design:

`fee = min(legacy_algo, 4844+alt-da-algo)`

the derivation pipeline is updated to accept an exclusive OR of calldata or blobs per tx, meaning that it can only have one or the other. It doesn't need to reject txs with both, it could ignore one if both are present

the batcher computes whether to use calldata or blobs depending on which is cheaper
we keep the overhead and the scalar as is and then add new fields for the 4844+alt-da-algo
the 4844+alt-da-algo works for both 4844 and alt-da and a newly added field similar to overhead/scalar can be used to parameterize the average % of blob used that is only consumed by the 4844+alt-da-algo

this gives us a very flexible design and keeps the complexity in the protocol encapsulated to a single area which is the min function, it isn't that much more complex than the existing fee algo. right now roberto is doing some crazy stuff to enable 1 or the other for fee calculation but still with a derivation pipeline that works for either meaning that gov would need to toggle between the two different fee charging schemes

**Pull request closed without merging**